node/lib/schema/OcfSchemaReader.ts

```diff
@@ -1,40 +1,380 @@
+     * Convert a parsed RAML API to a thing schema.
+     *
+     * Note this code (mostly) supports RAML v0.8 and v1.0 APIs.
+     * (OCF currently uses 0.8, so RAML 1.0 support is untested at this point.)
+     */
+    private static ramlToThingSchema(
+            ramlSchema: (raml.api08.Api | raml.api10.Api)): ThingSchema {
+        let schemaName = ramlSchema.title();
+        let description: string | undefined;
+        let methods: ThingMethod[] = [];
+
+        let ramlResources: Array<raml.api08.Resource | raml.api10.Resource> =
+                ramlSchema.resources();
+        ramlResources.forEach(ramlResource => {
+            // Remove a leading slash from the resource relative URI.
+            // Replace any other slashes, dots, or dashes with underscores.
+            let resourceName: string = ramlResource.relativeUri().value()
+                    .replace(/^\//, "").replace(/[/.-]/g, "_");
+
+            let resourceDescription: string = OcfSchemaReader.markdownToString(
+                    ramlResource.description());
+
+            // RAML doesn't have a top-level description, so use the description
+            // of the first resource as the thing schema description.
+            if (resourceDescription && !description) {
+                description = resourceDescription;
+            }
+
+            let ramlResourceMethods: Array<raml.api08.Method | raml.api10.Method> =
+                    ramlResource.methods();
+            ramlResourceMethods.forEach(ramlResourceMethod => {
+                let verb: string = ramlResourceMethod.method();
+                let methodDescription: string = OcfSchemaReader.markdownToString(
+                        ramlResourceMethod.description());
+
+                // Derive a JavaScript method name from the verb and resource name (URI).
+                let methodName: string = verb + OcfSchemaReader.capitalize(resourceName);
+
+                // Parameters are fixed, depending on the verb.
+                // Get the parameter types (JSON schemas) that are specified in the RAML.
+                let methodParameters: ThingParameter[] = [];
+                if (verb === "get") {
+                    let responseSchema: JsonSchema =
+                            OcfSchemaReader.getResourceMethodResponseSchema(
+                                ramlSchema.schemas(), resourceName, ramlResourceMethod);
+                    methodParameters.push({
+                        isOut: true,
+                        name: "result",
+                        parameterType: responseSchema,
+                    });
+                } else if (verb === "post") {
+                    let requestSchema: JsonSchema =
+                            OcfSchemaReader.getResourceMethodRequestSchema(
+                                ramlSchema.schemas(), resourceName, ramlResourceMethod);
+                    methodParameters.push({
+                        isOut: false,
+                        name: "value",
```

Comment:
I think the name of the parameter really doesn't matter. The tranlator .js code can name the parameter anything and the name is not going to be validated. (The type is what matters.)

---
In reply to: [75386316](https://github.com/openT2T/opent2t/pull/23#discussion_r75386316) [](ancestors = 75386316)